metrics formatting: preserve trailing whitespace

macfarla opened the pull request:
The newline character is required at the end of this string. 
Any other whitespace is ignored as per https://prometheus.io/docs/instrumenting/exposition_formats/ so no need to trim
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

I have tested this locally and get the following response from 
` curl localhost:8581/metrics`
Before ie without this change:
```

tessera_P2P_GET_getPartyInfo_MinTime_ms 0
tessera_P2P_GET_getPartyInfo_MaxTime_ms 0
...
tessera_THIRD_PARTY_GET_getVersion_RequestRate_requestsPerSeconds 0.0
tessera_THIRD_PARTY_GET_getVersion_RequestCount 0%
```
^ no new line at end of string

After ie with this change:
```
tessera_P2P_GET_getPartyInfo_MinTime_ms 0
tessera_P2P_GET_getPartyInfo_MaxTime_ms 0
...
tessera_THIRD_PARTY_GET_getVersion_RequestRate_requestsPerSeconds 0.0
tessera_THIRD_PARTY_GET_getVersion_RequestCount 0
```
so there is a newline at the end. Prometheus serves up the list of metrics.

macfarla committed:
no need to trim

Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

chris-j-h commented on server/jersey-server/src/main/java/com/quorum/tessera/server/monitoring/MetricsResource.java:
```diff
@@ -39,7 +39,7 @@ public Response getMetrics() {
     return Response.status(Response.Status.OK)
         .header(HttpHeaders.CONTENT_TYPE, TEXT_PLAIN)
-        .entity(formattedMetrics.toString().trim())
+        .entity(formattedMetrics.toString())
```
Instead of ensuring there's a single new line at the end of the response, this can actually result in multiple new lines (1 for each `AppType` that isn't in use).  Empty lines are fine but we can address this and simplify some things by removing the trim in `PrometheusProtocolFormatter`, which also means we can get rid of the `append("\n")` in `MetricsResource` i.e.

```diff
- formattedMetrics.append(formatter.format(metrics, type)).append("\n");
+ formattedMetrics.append(formatter.format(metrics, type)); 
```

chris-j-h replied:
I've disabled the slack notification step for the time being so this can be merged while we debug - i also moved the newline formatting into the PrometheusProtocolFormatter so that we can add some unit tests

macfarla committed:
remove trim and ensure \n at end

Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

chris-j-h committed:
Remove whitespace trim from PrometheusProtocolFormatter and unnecessary newline from API resp
The Prometheus spec defines that output must end with a newline, this change makes sure this is handled in the Formatter so that the Resource doesn't have to be concerned about the Prometheus spec

chris-j-h committed:
Disable failing slack notification steps so that builds can pass while debugging
